### PR TITLE
Added support for aggregate alerts

### PR DESCRIPTION
**Pull request by triceras**

### CODE REVIEW

File: examples/humioaggregatealert.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: core.humio.com/v1alpha1
+kind: HumioAggregateAlert
+metadata:
+    name: example-aggregate-alert-managed
+spec:
+  managedClusterName: example-humiocluster
+  name: example-aggregate-alert
+  queryString: "#repo = humio | error = true "
```
Comment (SaaldjorMike) [inline examples/humioaggregatealert.yaml:8]: Both examples should have an aggregate query as examples, so perhaps append ` | count()` to the query strings.

**Review by SaaldjorMike — CHANGES_REQUESTED** [review]
Awesome work. Mostly minor things, but they do need fixing before it is ready to get merged.

See my comments for various lines.

Also:
- we also have to update the operator-rbac.yaml file in the helm chart to add the new type to the 2 lists of the different CRD's we have
- this seems to throw some errors due to some fields not being set correctly:
```
Reason: 'queryTimestampType' Enum value '' is undefined in enum type 'QueryTimestampType'. Known values are: EventTimestamp, IngestTimestamp.

Reason: 'queryOwnershipType' Enum value '' is undefined in enum type 'QueryOwnershipType'. Known values are: User, Organization.
```
which causes aggregate alert creation to fail. Both filter alerts and scheduled search implementations already set the query ownership type, so can probably do the same here, and then also similar things to the timestamp type.
- finally, we have to decide what we want to do for the versioning. Aggregate alerts was introduced in 1.147.0 and tests currently run with older version than that. I ran a quick test with 1.150.0 using https://github.com/humio/humio-operator/pull/841/commits/78b76b9ef1051134aa36c3fe362dfc99d3ce1068 but that ran into the issue I mentioned about `QueryTimestampType` and `QueryOwnershipType`.

File: config/samples/core_v1alpha1_humioaggregatealert.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: core.humio.com/v1alpha1
+kind: HumioAggregateAlert
+metadata:
+  labels:
+    name: humioaggregatealert-sample
```
Comment (SaaldjorMike) [inline config/samples/core_v1alpha1_humioaggregatealert.yaml:5]:
```suggestion
  name: humioaggregatealert-sample
```

File: controllers/humiofilteralert_controller.go [hunk 3]
```diff
@@ -218,6 +218,5 @@ func (r *HumioFilterAlertReconciler) logErrorAndReturn(err error, msg string) er
 }
 
 func sanitizeFilterAlert(filterAlert *humioapi.FilterAlert) {
-	filterAlert.ID = ""
```
Comment (SaaldjorMike) [inline controllers/humiofilteralert_controller.go:221]: This was merged to master, so not sure if it is a merge of master into this branch/rebase gone wrong as it should not show up in this PR. You may need to do the same for aggregate alerts though, but removing this line and the same line for scheduled search is not supposed to be in this PR

File: pkg/humio/filteralert_transform.go [hunk 4]
```diff
@@ -27,6 +27,10 @@ func FilterAlertTransform(hfa *humiov1alpha1.HumioFilterAlert) (*humioapi.Filter
 		filterAlert.ID = hfa.ObjectMeta.Annotations[FilterAlertIdentifierAnnotation]
 	}
 
+	if filterAlert.Labels == nil {
+		filterAlert.Labels = []string{}
+	}
+
 	return filterAlert, nil
```
Comment (SaaldjorMike) [inline pkg/humio/filteralert_transform.go:34]: Same as the comment from on the changes to sanitizeFilterAlert. This should not be part of this PR. Same goes for updates to pkg/humio/scheduledsearch_transform.go

File: .github/workflows/e2e.yaml [hunk 5]
```diff
@@ -55,4 +55,4 @@ jobs:
         curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.22.0/kind-linux-amd64
         chmod +x ./kind
         ./kind delete cluster || true
-        make clean
+        make clean
```
Comment (SaaldjorMike) [inline .github/workflows/e2e.yaml:58]: You should clean up the state of this PR so that this is not included. It just creates unnecessary noise as this makes no real difference to the core work in this PR. Same goes for the update to .github/workflows/preview.yaml

File: controllers/humiofilteralert_controller.go (hunk 3 above)
Comment (triceras) [reply controllers/humiofilteralert_controller.go:221]: Not sure why it is showing up on this PR. Let me remove it.

File: pkg/humio/filteralert_transform.go (hunk 4 above)
Comment (triceras) [reply pkg/humio/filteralert_transform.go:34]: I do not have any idea why it is showing up on this PR. Perhaps I accidently edited the file. I remove it from the PR.

File: config/samples/core_v1alpha1_humioaggregatealert.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+apiVersion: core.humio.com/v1alpha1
+kind: HumioAggregateAlert
+metadata:
+  labels:
```
Comment (SaaldjorMike) [inline config/samples/core_v1alpha1_humioaggregatealert.yaml:4]:
```suggestion
```
if we dont have any labels, i dont see why we would have that here

File: pkg/humio/scheduledsearch_transform.go [hunk 7]
```diff
@@ -30,6 +30,10 @@ func ScheduledSearchTransform(hss *humiov1alpha1.HumioScheduledSearch) (*humioap
 		scheduledSearch.ID = hss.ObjectMeta.Annotations[ScheduledSearchIdentifierAnnotation]
 	}
 
+	if scheduledSearch.Labels == nil {
+		scheduledSearch.Labels = []string{}
+	}
+
```
Comment (SaaldjorMike) [inline pkg/humio/scheduledsearch_transform.go:36]: Not related to this PR. Not sure how this got here

**Review by SaaldjorMike — APPROVED** [review]
LGTM. Looks like the only failing tests we have is related to the concurrency issue with the mock, but https://github.com/humio/humio-operator/pull/843 will refactor how that works and https://github.com/humio/humio-operator/pull/843 will be updated to fix this for aggregate alerts as well.
